Extensions/ArtifactEngine/Engine/cilogger.ts

```diff
@@ -1,21 +1,45 @@
+import { error } from "util";
+
 var tl = require('vsts-task-lib');
+var packagejson = require('../package.json');
 
 const area: string = 'artifact-engine';
 
 function getDefaultProps() {
     return {
-        releaseuri: tl.getVariable('Release.ReleaseUri'),
+        serverurl: tl.getVariable('System.TEAMFOUNDATIONSERVERURI'),
+        releaseurl: tl.getVariable('Release.ReleaseWebUrl'),
         releaseid: tl.getVariable('Release.ReleaseId'),
         builduri: tl.getVariable('Build.BuildUri'),
-        buildid: tl.getVariable('Build.Buildid')
+        buildid: tl.getVariable('Build.Buildid'),
+        agentVersion: tl.getVariable('Agent.Version'),
+        version: packagejson.version
     };
 }
 
-export function publishEvent(feature, properties: { [key: string]: any }): void {
+export function publishEvent(feature, properties: any | IReliabilityData): void {
     try {
-        tl.assertAgent('2.125.0');
-        tl.publishTelemetry(area, feature, Object.assign(getDefaultProps(), properties));
-    } catch (err) {
-        tl.debug('Unable to publish telemetry due to lower agent version.');
+        var splitVersion = (process.env.AGENT_VERSION || '').split('.');
+        var major = parseInt(splitVersion[0] || '0');
+        var minor = parseInt(splitVersion[1] || '0');
+        let telemetry = '';
+        if (major > 2 || (major == 2 && minor >= 120)) {
+            telemetry = `##vso[telemetry.publish area=${area};feature=${feature}]${JSON.stringify(Object.assign(getDefaultProps(), properties))}`;
+        }
+        else {
+            if (feature === 'reliability') {
+                let reliabilityData = <IReliabilityData>properties;
+                telemetry = "##vso[task.logissue type=error;code=" + reliabilityData.issueType + ";agentVersion=" + tl.getVariable('Agent.Version') + ";taskId=" + area + "-" + packagejson.version + ";]" + reliabilityData.errorMessage
+            }
+        }
+        console.log(telemetry);;
```

Comment:
i thought you started using task library to log to console. should we use task library here instead?

Reply:
task lib is used for debug and warning messages to log normal messages we need console.log